src/index.ts

```diff
@@ -159,30 +222,82 @@ class Compiler {
       case RuleType.AnyOf:
         return new TsType.Union(rule.anyOf!.map(_ => this.toTsType(_)))
       case RuleType.Reference:
-        return this.resolveType(rule.$ref!)
+        return this.resolveType(rule.$ref!, propName!)
     }
     throw new Error('Unknown rule:' + rule.toString())
   }
-  private toTsType(rule: JSONSchema.Schema): TsType.TsType {
-    let type = this.createTsType(rule)
+
+  private validateEnumMembers(rule: JSONSchema.Schema): EnumType {
+    if (!rule.type) rule.type = 'string'
+
+    let isDeclaredStringEnum = rule.type === 'string'
+    let isDeclaredIntegerEnum = rule.type === 'integer'
+
+    if (!isDeclaredStringEnum && !isDeclaredIntegerEnum){
+      throw TypeError('Enum type must be string or integer; default is string if undefined')
+    }
+
+    if (rule.enum!.some(_ => _ instanceof Object)){
```

Comment:
We already sort of do that.  If we are declared as something other than type=string or type=integer, we have already returned by now.  

We then follow this check with other checks for the general "not string and not int" case.  The only consequence of removing this line would be a slightly-less-specific error message.